darwin/datetimepicker.m

```diff
@@ -40,3 +120,36 @@
 {
 	return finishNewDateTimePicker(NSHourMinuteSecondDatePickerElementFlag);
 }
+
+void uiDateTimePickerTime(uiDateTimePicker *d, struct tm *time)
+{
+	time_t t;
+	struct tm tmbuf;
+	NSDate *date;
+	NSTimeInterval interval;
+
+	date = [d->dp dateValue];
+	interval = [[NSTimeZone systemTimeZone] secondsFromGMTForDate:date];
```

Comment:
What do this line and the next one do? I'm not sure if I'm comfortable using NSTimeInterval since that's really a double and (from what I can gather) `time_t` is not; what about `NSDateComponents` (with `[d->dp calendar]` as the source calendar to extract the components with)? I forget if `NSDate` has something more precise than `NSTimeInterval` as an option (maybe Core Foundation does via toll-free bridging?)...

Reply:
Honestly have no idea why the `NSTimeZone` bit is needed! It makes the example program work. Without it, the dates returned are totally wrong. (`NSDatePicker` is weird!)

Agreed `double` to `time_t` conversation looks really bad. There is no portable size guarantee for `time_t`. 
Though I think `NSTimeInterval` can be truncated to an integer type okay since we only need resolution to the second.

I'll have to research more.. and definitely could use a bit of help on these two.

Reply:
Honestly have no idea why the `NSTimeZone` bit is needed! It makes the example program work. Without it, the dates returned are totally wrong. (`NSDatePicker` is weird!)

Agreed `double` to `time_t` conversation looks really bad. There is no portable size guarantee for `time_t`. Though I think `NSTimeInterval` can be truncated to an integer type okay since we only need resolution to the second.

I'll have to research more.. and definitely could use a bit of help on these two.

Reply:
Fixed the `NSTimeZone` nonsense!

`NSTimeInterval` to `time_t` maybe still an issue.